platform/macos/export/export_plugin.cpp

```diff
@@ -395,9 +392,17 @@ void EditorExportPlatformMacOS::get_export_options(List<ExportOption> *r_options
 	r_options->push_back(ExportOption(PropertyInfo(Variant::STRING, "xcode/xcode_build"), "14C18"));
 
 #ifdef MACOS_ENABLED
-	r_options->push_back(ExportOption(PropertyInfo(Variant::INT, "codesign/codesign", PROPERTY_HINT_ENUM, "Disabled,Built-in (ad-hoc only),rcodesign,Xcode codesign"), 3, true));
+	if (Engine::get_singleton()->has_singleton("GodotSharp")) {
+		r_options->push_back(ExportOption(PropertyInfo(Variant::INT, "codesign/codesign", PROPERTY_HINT_ENUM, "Disabled,Built-in (ad-hoc only),Xcode codesign:3"), 3, true));
```

Comment:
What about existing export presets that are already set to `2`?

Reply:
Maybe it's simpler to keep the same enum for all builds, but just show the error added below if a user selects rcodesign for a C# project. It's not the best UX but it should simplify things and not be that bad either (at least not worse than a lot of the export settings UX).

Reply:
Changed it to a warning.